### PR TITLE
Networking params

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- allow to specify Network IP ranges: worker CIDR (nodes CIDR), Pods CIDR, services CIDR
- default nodes CIDR: 10.250.0.0/22
- ...

**Related issue(s)**

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
can be merged when https://github.com/kyma-project/control-plane/pull/2968 is merges

File: components/kyma-environment-broker/internal/provider/aws_provider.go [hunk 1]
```diff
@@ -121,54 +124,105 @@ func MultipleZonesForAWSRegion(region string, zonesCount int) []string {
...
+	//    PublicCidr:   "10.250.32.0/20",
+	//    InternalCidr: "10.250.48.0/20",
+	// 4 * delta  - difference between two worker (zone) CIDRs
+	delta := big.NewInt(1)
+	delta.Lsh(delta, uint(31-workerPrefixLength))
+
+	// base - it is an integer, which is based on IP bytes
+	base := new(big.Int).SetBytes(workerPrefix.Addr().AsSlice())
+
+	for _, name := range zoneNames {
```
Comment (jaroslaw-pieszka) [inline components/kyma-environment-broker/internal/provider/aws_provider.go:163]: Is number of zones limited somehow?

File: components/kyma-environment-broker/internal/provider/azure_provider.go [hunk 2]
```diff
@@ -249,15 +276,29 @@ func generateRandomAzureZones(zonesCount int) []int {
 	return zones[:zonesCount]
 }
 
-func generateMultipleAzureZones(zoneNames []int) []*gqlschema.AzureZoneInput {
-	subnetFmt := "10.250.%d.0/19"
-	zones := []*gqlschema.AzureZoneInput{}
-	for i, zone := range zoneNames {
+func generateAzureZones(workerCidr string, zoneNames []int) []*gqlschema.AzureZoneInput {
```
Comment (jaroslaw-pieszka) [inline components/kyma-environment-broker/internal/provider/azure_provider.go:279]: Are we sure that there won't be more zones than 2?

File: components/kyma-environment-broker/internal/provider/aws_provider.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/provider/aws_provider.go:163]: For 6 zones (6 iterations) we will create the zone outside the CIDR from parameter `workerCidr`

File: components/kyma-environment-broker/internal/broker/instance_create.go [hunk 3]
```diff
@@ -427,3 +440,68 @@ func (b *ProvisionEndpoint) createDashboardURL(planID, instanceID string) string
...
+	if parameters.Networking == nil {
+		return nil
+	}
+	var nodes, services, pods *net.IPNet
+	if nodes, e = validateCidr(parameters.Networking.NodesCidr); e != nil {
+		err = multierror.Append(err, fmt.Errorf("while parsing nodes CIDR: %w", e))
+	}
+	// error is handled before, in the validate CIDR
+	cidr, _ := netip.ParsePrefix(parameters.Networking.NodesCidr)
+	if cidr.Bits() > 26 {
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/broker/instance_create.go:473]: The maximum Bits we should support is /24, because when going with /24 nodes allows assigning 3x /26 subnet for the 3 worker subnets, while there are still 8x /28 subnets, 6x of which would be used to assign to public and internal ELB subnets.

However to support this scenario, you would also need to adjust the hard-coded `workerPrefixLength := cidr.Bits() + 3` logic, as when /24 was input, you would need to go with `workerPrefixLength := cidr.Bits() + 2`, and `workerPrefixLength + 2` for `public` and `internal`

TL;DR: in general, the smallest CIDR for any workers subnet per AZ should be /26 , as it allows 62 nodes. /27 would be too low already. For Azure, we could even split /24 easily into 4x /26, but for AWS, it's trickier than the current implementation in aws_provider.go

**Review by ebensom — CHANGES_REQUESTED** [review]

File: components/kyma-environment-broker/internal/broker/instance_create.go (hunk 3 above)
Comment (ebensom) [reply components/kyma-environment-broker/internal/broker/instance_create.go:473]: Or do you think if we explicitly document to the user than when providing the smallest possible /26 nodes CIDR, the runtime will only be able to scale up to maximum 6 nodes per AZ (18 nodes total) using /29 CIDR?, And furthermore on AWS they will only be able to create 2 public LoadBalancer Services (because of the /30 public CIDR per AZ)?

I think this is a big limitation though

Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/broker/instance_create.go]: I'll add a validation - max value in the suffix is 23

**Review by jaroslaw-pieszka — DISMISSED** [review]

File: components/kyma-environment-broker/internal/broker/instance_create.go (hunk 3 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/broker/instance_create.go]: let's do it +2 and suffix 24

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by ebensom — APPROVED** [review]
